src/spikeinterface/extractors/neoextractors/plexon.py

```diff
@@ -15,22 +15,41 @@ class PlexonRecordingExtractor(NeoBaseRecordingExtractor):
 
     Parameters
     ----------
-    file_path : str
+    file_path : str | Path
         The file path to load the recordings from.
     stream_id : str, default: None
         If there are several streams, specify the stream id you want to load.
     stream_name : str, default: None
         If there are several streams, specify the stream name you want to load.
     all_annotations : bool, default: False
         Load exhaustively all annotations from neo.
+    use_names_as_ids : bool, default: True
+        If True, the names of the signals are used as channel ids. If False, the channels ids are the ids provided
```

Comment:
Is this name of the signal or channel_names? In neo we would call them channel_names, but do you thing signals is more parseable for our spikeinterface users?

Reply:
`channel_names` check the name of the field in the structure array for signals that I shared below and the code:

https://github.com/h-mayorquin/spikeinterface/blob/006f98c9b07dcac07cdf58efb914aba5642d9bf1/src/spikeinterface/extractors/neoextractors/neobaseextractor.py#L233-L241

Reply:
I mean in the docstring is the word signals the best word or do we say the Neo term and say use the channel_names as provided by Neo. Based on the neo issue we've got linked I don't think semantically our name vs id vs index is super clear similar to how the `stream` semantic is a bit hard to follow. So what is the best way for this type of docstring to clearly convey what a channel_name means so people can honestly make a choice. 

For example for intan the difference between name and id is actually the difference between using the custom user name and the hardware native name. Here for plexon it is the difference between an actual hardware channel name and a stringified index. So there isn't really a unified meaning to `use_names_as_ids` to be clear for our end-users.

Reply:
But to be clear this shouldn't slow down this PR, but I think we need to come up with a better way to explain this.

Reply:
No, no, I think you are right. Sorry I misread.  Yes, the docstring should be changed. Given that there is no unified semantics we need something better. Let me make a commit and see what you think. It is good to discuss that because I am enabling this attribute for all extractors (not changing the default behavior like here though) so users don't need to wait for us if they want to choose.

Give me a sec.

Reply:
Check the new one. in the latest commit. It does not make any promises and is quite literal. Then, when we know (as it takes some work and knowledge of the formats) we can add clarifications about the semantics and example differences as we do here.

So the template of the docstring is:
1) The abstract "I don't make any claims" think here.
2) Some clarification about what each of them are (for example in Intan we can mentioned what is what).
3) An example of how they look like.

What do you think?

Reply:
Yeah, I think that is best. It's a lot of work for all neo.rawios, but I guess we are little stuck by the nature of the neo names vs ids. Thanks!

Reply:
Yeah, I am gonga propagate everything with 1. And hopefully we can address 2 and 3 slowly but surely.

Gradatim Ferociter

Reply:
Festina lente, amice.